File: LeetCodeProblemsGo/LeetCode3021AliceAndBobPlayingFlowerGame/LeetCode3021AliceAndBobPlayingFlowerGame.go
package LeetCode3021AliceAndBobPlayingFlowerGame

func flowerGame(n int, m int) int64 {
	var firstEven int64 = int64(n / 2)
	var firstOdd int64 = int64(n - n/2)
	var secondEven int64 = int64(m / 2)
	var secondOdd int64 = int64(m - m/2)
	return firstEven*secondOdd + secondEven*firstOdd
}

File: LeetCodeProblemsCMake/Problems/LeetCode3021AliceAndBobPlayingFlowerGame.cpp
//
// Created by saeed on 8/29/25.
//

#include "LeetCode3021AliceAndBobPlayingFlowerGame.h"

long long LeetCode3021AliceAndBobPlayingFlowerGame::flowerGame(int n, int m) {
    long long result {0};
    long long firstEven=n/2;
    long long firstOdd=n-n/2;
    long long secondEven=m/2;
    long long secondOdd=m-m/2;
    result= firstEven*secondOdd+firstOdd*secondEven;
    return result;
}
